unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PlayerName/PlayerName.cs

```diff
@@ -72,18 +100,23 @@ internal void Update(float deltaTime)
             UpdateVisuals(0);
             return;
         }
-        
+
+        float previousAlphaStep = GetNearestAlphaStep(alpha);
         float finalTargetAlpha = forceShow ? TARGET_ALPHA_SHOW : targetAlpha;
+        alpha = Mathf.MoveTowards(alpha, finalTargetAlpha, ALPHA_TRANSITION_STEP_PER_SECOND * deltaTime);
+        float currentAlphaStep = GetNearestAlphaStep(alpha);
+        if (currentAlphaStep == previousAlphaStep)
+            return;
```

Comment:
This return is going to cause some issues regarding scaling and rotation, you might want to raise transform calls above alpha or you can also extract all of this logic into a new Method for more clarity. `UpdateVisuals` is a good candidate to have this logic also.